tests/components/bmw_connected_drive/test_config_flow.py

```diff
@@ -0,0 +1,164 @@
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.bmw_connected_drive.config_flow import DOMAIN
+from homeassistant.components.bmw_connected_drive.const import (
+    CONF_READ_ONLY,
+    CONF_REGION,
+    CONF_USE_LOCATION,
+)
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.async_mock import patch
+from tests.common import MockConfigEntry
+
+FIXTURE_USER_INPUT = {
+    CONF_USERNAME: "user@domain.com",
+    CONF_PASSWORD: "p4ssw0rd",
+    CONF_REGION: "rest_of_world",
+}
+FIXTURE_COMPLETE_ENTRY = FIXTURE_USER_INPUT.copy()
+FIXTURE_IMPORT_ENTRY = FIXTURE_USER_INPUT.copy()
+
+FIXTURE_CONFIG_ENTRY = {
+    "entry_id": "1",
+    "domain": DOMAIN,
+    "title": FIXTURE_USER_INPUT[CONF_USERNAME],
+    "data": {
+        CONF_USERNAME: FIXTURE_USER_INPUT[CONF_USERNAME],
+        CONF_PASSWORD: FIXTURE_USER_INPUT[CONF_PASSWORD],
+        CONF_REGION: FIXTURE_USER_INPUT[CONF_REGION],
+    },
+    "options": {CONF_READ_ONLY: False, CONF_USE_LOCATION: False},
+    "system_options": {"disable_new_entities": False},
+    "source": "user",
+    "connection_class": config_entries.CONN_CLASS_CLOUD_POLL,
+    "unique_id": f"{FIXTURE_USER_INPUT[CONF_REGION]}-{FIXTURE_USER_INPUT[CONF_REGION]}",
+}
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_auth_error(hass):
+    """Test we show user form on BMW connected drive connection error."""
+
+    with patch(
+        "bimmer_connected.account.ConnectedDriveAccount._get_oauth_token",
+        side_effect=OSError,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=FIXTURE_USER_INPUT,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "invalid_auth"}
+
+
+async def test_connection_error(hass):
+    """Test we show user form on BMW connected drive connection error."""
+
+    with patch(
+        "bimmer_connected.account.ConnectedDriveAccount._get_oauth_token",
+        side_effect=Exception,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data=FIXTURE_USER_INPUT,
+        )
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+    assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_general_error(hass):
+    """Test we show user form on BMW connected drive connection error."""
+
+    with patch(
+        "homeassistant.config_entries.ConfigEntry",
```

Comment:
This patch is incorrect. Creating the config entry can't error.

Reply:
Removed this whole test as we're testing for a global exception above.